pkg/ingress/kube/ingress/controller.go

```diff
@@ -1243,48 +1230,60 @@ func setDefaultMSEIngressOptionalField(ing *ingress.Ingress) {
 }
 
 // createRuleKey according to the pathType, path, methods, headers, params of rules
-func createRuleKey(annots map[string]string, hostAndPath string) (uint32, string, error) {
+func createRuleKey(annots map[string]string, hostAndPath string) string {
 	var (
-		headers []string
-		params  []string
+		headers [][2]string
+		params  [][2]string
 		sb      strings.Builder
+		sep     byte = '\n'
 	)
 
 	// path
 	sb.WriteString(hostAndPath)
+	sb.WriteByte(sep)
 
 	// methods
 	if str, ok := annots[annotations.HigressAnnotationsPrefix+"/"+annotations.MatchMethod]; ok {
 		sb.WriteString(str)
 	}
+	sb.WriteByte(sep)
+
+	start := len(annotations.HigressAnnotationsPrefix) + 1 // example: higress.io/exact-match-header-key: value
 	// headers && params
-	for k, _ := range annots {
+	for k, val := range annots {
 		if idx := strings.Index(k, annotations.MatchHeader); idx != -1 {
-			headers = append(headers, k)
+			key := k[start:idx] + k[idx+len(annotations.MatchHeader)+1:]
+			headers = append(headers, [2]string{key, val})
 		}
 		if idx := strings.Index(k, annotations.MatchQuery); idx != -1 {
-			params = append(params, k)
+			key := k[start:idx] + k[idx+len(annotations.MatchQuery)+1:]
+			params = append(params, [2]string{key, val})
 		}
 	}
 	sort.SliceStable(headers, func(i, j int) bool {
-		return headers[i] < headers[j]
+		return headers[i][0] < headers[j][0]
 	})
 	sort.SliceStable(params, func(i, j int) bool {
-		return params[i] < params[j]
+		return params[i][0] < params[j][0]
 	})
 	for idx := range headers {
-		sb.WriteString(headers[idx])
-		sb.WriteString(annots[headers[idx]])
+		if idx != 0 {
+			sb.WriteByte('\f')
```

Comment:
The character ‘\f’ is not debug friendly

Reply:
done